vyper/builtin_functions/functions.py

```diff
@@ -2576,6 +2568,25 @@ def _eval_decimal(self, type_):
         return typinfo.decimal_bounds[1]
 
 
+class Epsilon(FoldedFunction):
+
+    _inputs = [("typename", "TYPE_DEFINITION")]
+    _id = "epsilon"
+
+    def evaluate(self, node):
+        self._validate_arg_types(node)
+        input_type = get_type_from_annotation(node.args[0], DataLocation.UNSET)
+
+        if not isinstance(input_type, FixedAbstractType):
+            raise InvalidType(f"Expected decimal type but got {input_type} instead", node)
+
+        # this check seems redundant, but sets a pattern to be followed
+        # when new decimal types are created
+        if isinstance(input_type, DecimalDefinition):
+            typinfo = parse_decimal_info(str(input_type))
+            return vy_ast.Decimal.from_node(node, value=typinfo.epsilon)
+
```

Comment:
probably add a `raise CompilerPanic("unreachable")` at the end of the control flow here